src/main/java/org/trigon/hardware/rev/spark/io/SimulationSparkIO.java

```diff
@@ -63,51 +73,23 @@ public void setInverted(boolean inverted) {
     }
 
     @Override
-    public void enableVoltageCompensation(double voltage) {
-        motor.enableVoltageCompensation(voltage);
+    public void updateSimulation() {
+        if (motorSimulation == null)
+            return;
+        absoluteEncoderSimulation.iterate(encoder.getVelocityRotationsPerSecond(), RobotHardwareStats.getPeriodicTimeSeconds());
+        motorSimulation.iterate(absoluteEncoderSimulation.getVelocity(), 12, RobotHardwareStats.getPeriodicTimeSeconds());
+        physicsSimulation.updateMotor();
+        physicsSimulation.setInputVoltage(motorSimulation.getBusVoltage());
```

Comment:
move the magic number to a constant in RobotHardwareStats (also from the other sim that does this). 
try to repeat the logic in the TalonFX sim, it is really similar.